File: src/test/java/rzaevali/ScheduleTest.java
package rzaevali;

import com.google.common.collect.ImmutableList;
import org.junit.Test;
import rzaevali.exceptions.UnknownValueException;
import rzaevali.exceptions.VyatsuScheduleException;
import rzaevali.utils.JsonUtils;
import rzaevali.utils.PdfUtils;
import rzaevali.utils.ScheduleUtilsKt;

import java.io.FileReader;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import java.util.Map;

import static org.junit.Assert.assertEquals;

public class ScheduleTest {

    private static final String INCORRECT_SEASON = "null";

    private static final String TEST_GROUP_ID = "0000";

    @Test
    public void testParsing() throws IOException, VyatsuScheduleException {
        List<Integer> groupIds = ImmutableList.of(
                7795, 7794, 8891, 8969
        );

        String pathPattern = "src/test/resources/%s/%s_1_18122017_31122017.%s";

        for (Integer groupId : groupIds) {
            String pdfPath = String.format(pathPattern, "pdf", groupId, "pdf");
            String jsonPath = String.format(pathPattern, "json", groupId, "json");

            List data = PdfUtils.INSTANCE.extractSchedule(Files.newInputStream(Paths.get(pdfPath)));
            List originalData = (List) JsonUtils.getDefaultJson()
                    .fromJson(new FileReader(jsonPath), Map.class).get("weeks");

            assertEquals(Integer.toString(groupId), originalData, data);
        }
    }

    @Test(expected = UnknownValueException.class)
    public void testArgumentChecks() throws VyatsuScheduleException {
        ScheduleUtilsKt.getSchedule(TEST_GROUP_ID, INCORRECT_SEASON);
    }

}

File: src/test/kotlin/rzaevali/ScheduleTest.kt
package rzaevali

import com.google.common.collect.ImmutableList
import com.google.gson.GsonBuilder
import org.junit.Assert.assertEquals
import org.junit.Test
import rzaevali.exceptions.UnknownValueException
import rzaevali.exceptions.VyatsuScheduleException
import rzaevali.utils.extractSchedule
import rzaevali.utils.getSchedule
import java.io.FileReader
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Paths

class ScheduleTest {

    @Test
    @Throws(IOException::class, VyatsuScheduleException::class)
    fun testParsing() {
        val gson = GsonBuilder().create()

        val groupIds = ImmutableList.of(
                7795, 7794, 8891, 8969
        )

        val pathPattern = "src/test/resources/%s/%s_1_18122017_31122017.%s"

        for (groupId in groupIds) {
            val pdfPath = String.format(pathPattern, "pdf", groupId, "pdf")
            val jsonPath = String.format(pathPattern, "json", groupId, "json")

            val data = extractSchedule(Files.newInputStream(Paths.get(pdfPath)))
            val originalData = gson.fromJson(FileReader(jsonPath), Map::class.java)["weeks"] as List<*>

            assertEquals(Integer.toString(groupId!!), originalData, data)
        }
    }

    @Test(expected = UnknownValueException::class)
    @Throws(VyatsuScheduleException::class)
    fun testArgumentChecks() {
        getSchedule(TEST_GROUP_ID, INCORRECT_SEASON)
    }

    companion object {

        private val INCORRECT_SEASON = "null"

        private val TEST_GROUP_ID = "0000"

    }

}
